### PR TITLE
Support for building using jdk 17, Gradle wrapper upgrade to 7.6 version  and Copyright update

### PR DESCRIPTION
**Pull request by bmscomp**
This pull requests aims to mainly upgrade Gradle to version 7.6 to support building the project with the new version of the wrapper and new version of java as JDK 17 and JDK 19, the pull request is widely inspired by the great job done on vavr project, there is also an update of Copyright to make it valid till the current new year 2023 

- upgrade Gradle wrapper version to 7.6 
- update .gitignore file to ignore visual studio and eclipse files and swap files 
- copyright update to 2023
- add jdk 17 and 19 to travis.yml 
- remove deprecation call of new Integer(int)
- use a compatible publish plugin with Gradle 7.6 
- upgrade jupiter junit version to 5.9.1
- upgrade Jaxb dependency to 2.3.1

### CODE REVIEW

**Review by mincong-h — COMMENTED** [review]
Thanks a lot for preparing this PR @bmscomp , really happy to see you as a contributor here! And thanks for taking extra mile to test the publishing process 😁 I put two comments here but the PR LGTM in general

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,7 @@
-junitJupiterVersion=5.7.0
-vavrVersion=1.0.0-SNAPSHOT
+junitJupiterVersion=5.9.1
+vavrVersion=1.0.0-alpha-4
```
[the remark this answers is not in the document]
Comment (bmscomp) [reply gradle.properties:2]: You'r right , I will restore the previous version  :)

File: .travis.yml [hunk 2]
```diff
@@ -4,10 +4,13 @@ jdk:
   - oraclejdk8
   - openjdk11
   - openjdk14
+  - openjdk17
+  - openjdk19
```
[the remark this answers is not in the document]
Comment (bmscomp) [reply .travis.yml:8]: I can give it a try to move forward Github actions :)

**Comment by ruslansennov** [comment]
Hi @bmscomp @mincong-h 
Let's merge this PR and then I'll try to fix the credentials if need be

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,35 +1,58 @@
...
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.6
+
+      - name: Execute Gradle build
+        run: ./gradlew build
```
Comment (mincong-h) [inline .github/workflows/main.yml:33]: I am not quite sure the purpose of gradle build here. Isn't grade check include compilation (which builds everything) already?

**Comment by bmscomp** [comment]
Hi @mincong-h  @ruslansennov In this branch I added a publish job, but seems have an issue with credentials as in publish package step I got this error : ` Could not PUT 'https://oss.sonatype.org/content/repositories/snapshots/io/vavr/vavr-jackson/1.0.0-SNAPSHOT/maven-metadata.xml'. Received status code 401 from server: Unauthorized`, I used to think that we need some setup for username and passport to access sonatype repository and also a signing key and secret 



    name: "Publish"
    runs-on: ubuntu-latest
    needs: [ test ]
    steps:
      - uses: actions/checkout@v3
      - name: Set up Java
        uses: actions/setup-java@v3
        with:
          distribution: temurin
          java-version: 17
      - name: Publish package
        run: gradle -Prelease publishToSonatype closeAndReleaseSonatypeStagingRepository
        env:
          ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.SONATYPE_USER }}
          ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.SONATYPE_PASS }}
          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.SIGNING_KEY }}
          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.SIGNING_SECRET }} `

**Review by mincong-h — APPROVED** [review]

**Comment by mincong-h** [comment]
> Hi @mincong-h @ruslansennov In this branch I added a publish job, but seems have an issue with credentials as in publish package step I got this error : ` Could not PUT 'https://oss.sonatype.org/content/repositories/snapshots/io/vavr/vavr-jackson/1.0.0-SNAPSHOT/maven-metadata.xml'. Received status code 401 from server: Unauthorized`, I used to think that we need some setup for username and passport to access sonatype repository and also a signing key and secret
> 
> ```
> name: "Publish"
> runs-on: ubuntu-latest
> needs: [ test ]
> steps:
>   - uses: actions/checkout@v3
>   - name: Set up Java
>     uses: actions/setup-java@v3
>     with:
>       distribution: temurin
>       java-version: 17
>   - name: Publish package
>     run: gradle -Prelease publishToSonatype closeAndReleaseSonatypeStagingRepository
>     env:
>       ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.SONATYPE_USER }}
>       ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.SONATYPE_PASS }}
>       ORG_GRADLE_PROJECT_signingKey: ${{ secrets.SIGNING_KEY }}
>       ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.SIGNING_SECRET }}
> ```

Ruslan will help us handle that

File: .github/workflows/main.yml (hunk 3 above)
Comment (bmscomp) [reply .github/workflows/main.yml:33]: Yes the check is already building with various JDK versions, the build here is build on various operating systems and JDK versions also, agree that there is not much value on it, but as someone who have macOs I can see if it's builds on that operating system or not, @mincong-h do you think I should keep only one operating system and remove this step ?

Comment (mincong-h) [reply .github/workflows/main.yml:33]: Let's keep what you currently have. I am just asking out of curiosity. Having macOS build LGTM

**Comment by mincong-h** [comment]
Thanks for your contribution @bmscomp !! 😁

**Pull request merged**